### ISSUE TITLE
Refactor AuthorizationsUtil class

### ISSUE THREAD

**Issue by keith-ratcliffe** [NationalSecurityAgency/datawave#279]
AuthorizationsUtil's "downgrade" functionality is needed by all query-related components that are in the process of being transitioned to microservices. 

Therefore, we'll relocate those bits from AuthorizationsUtil.java to a new home under the services parent, and we'll refactor the existing class as needed

**Comment by keith-ratcliffe** [comment]
@brianloss , @tomnelson

What do you guys think about the following, for revising the UserAuthFunctions api from the last PR

First, define the following (somewhere)...
```
@FunctionalInterface
interface AuthSupplier {
    Collection<String> getAuths();
}
```
...and then just have DatawaveUser implement that interface. Alternatively, we could forego the new FunctionalInterface above, and just have ...
```
class DatawaveUser implements Supplier<Collection<String>>, ...
...
Collection<String> get() {
   return this.getAuths();
}
``` 
...but that seems rather confusing

Anyway, in AuthorizationsUtil, we could still just pass the DatawaveUser instance as before. That could eliminate the authorization-api dependency from accumulo-utils and still allow custom impl's of UserAuthFunctions to access our DatawaveUser instance(s) via explicit casting, if needed.

So, the interface methods could become....
```
Authorizations getRequestedAuthorizations(String requestedAuths, AuthSupplier authSupplier);
Set<Authorizations> mergeAuthorizations(Authorizations primaryUserAuths, Collection<AuthSupplier> proxyChain);
```

Thoughts?

**Comment by brianloss** [comment]
I don't think adding the FunctionalInterface buys anything, does it? Since we'd be passing the object and not a method reference, any old interface would work. That's certainly an option.

**Comment by keith-ratcliffe** [comment]
Yep, buys us nothing. Just used to demo comparison with the ```implements Supplier<>``` option, since ```Supplier``` itself is annotated as such

Really, just a matter of where something like AuthSupplier should live. *Could* be defined in accumulo-utils, I suppose, which then forces authorization-api to pick up accumulo-utils as a dependency. But perhaps that's preferable to the current relationship(?) Not sure. Or define yet another new module, where only interfaces and the like will reside(?) I dunno

**Comment by SethSmucker** [comment]
AuthorizationsUtil and the downgrade logic were pulled out to core/utils/common-utils and are shared by the microservices now, closing.
